### PR TITLE
Added OCP-26822 (duplicate of OCP-19810) but works on 4.x

### PR DESCRIPTION
**Pull request by anuragthehatter**
Automated this manual intensive case on 4.x.
@zhaozhanqi @huiran0826 @weliang1 @rbbratta 
Created new test case for it with correct version and moved OCP-19810 to 3.x
This  case is leveraging Internal IP of node def created in #567 
Passing Logs: http://pastebin.test.redhat.com/824391

Note: OVN has a bug for this usecase https://bugzilla.redhat.com/show_bug.cgi?id=1787434 however scenario is automated now keeping compaitibility with both plugins.

### CODE REVIEW

File: features/networking/pod.feature [hunk 1]
```diff
@@ -298,3 +298,89 @@ Feature: Pod related networking scenarios
...
+      | resource      | pod                      |
+      | resource_name | <%= cb.host_pod1.name %> |
+      | type          | NodePort                 |
+      | port          | 8080                     |
+      | protocol      | UDP                      |
+    Then the step should succeed
+    #Getting nodeport value
+    When I run the :get client command with:
+      | resource | service                                       |
+      | output   | jsonpath={.items[*].spec.ports[*].nodePort} |
```
Comment (huiran0826) [inline features/networking/pod.feature]: indention

File: features/networking/pod.feature [hunk 2]
```diff
@@ -298,3 +298,89 @@ Feature: Pod related networking scenarios
...
+      | name=hello-pod |
+    And evaluation of `pod` is stored in the :client_pod clipboard
+    # 'yes' command will send a character "h" continously for 3 seconds to /dev/udp on listener where the node is listening for udp traffic on exposed nodeport. The 3 seconds mechanism will create an Assured
+    #  entry which will give us enough time to validate upcoming steps
+    When I run the :exec background client command with:
+      | pod              | <%= cb.client_pod.name %>                              |
+      | oc_opts_end      |                                                        |
+      | exec_command     | bash                                                   |
+      | exec_command_arg | -c                                                     |
+      | exec_command_arg | yes "h">/dev/udp/<%= cb.node_ip %>/<%= cb.nodeport %> |
```
Comment (huiran0826) [inline features/networking/pod.feature]: indention

File: features/networking/pod.feature [hunk 3]
```diff
@@ -298,3 +298,89 @@ Feature: Pod related networking scenarios
...
+    And evaluation of `pod` is stored in the :host_pod2 clipboard
+
+    # 'yes' command will send a character "h" continously for 3 seconds to /dev/udp on listener where the node is listening for udp traffic on exposed nodeport. The 3 seconds mechanism will create an Assured
+    #  entry which will give us enough time to validate upcoming steps
+    When I run the :exec background client command with:
+      | pod              | <%= cb.client_pod.name %>                              |
+      | oc_opts_end      |                                                        |
+      | exec_command     | bash                                                   |
+      | exec_command_arg | -c                                                     |
+      | exec_command_arg | yes "h">/dev/udp/<%= cb.node_ip %>/<%= cb.nodeport %> |
```
Comment (huiran0826) [inline features/networking/pod.feature]: indention

**Comment by anuragthehatter** [comment]
> /lgtm except the indention

done

File: features/networking/pod.feature [hunk 4]
```diff
@@ -298,3 +298,89 @@ Feature: Pod related networking scenarios
...
+      | name=network-pod |
+    And evaluation of `pod.name` is stored in the :network_pod clipboard
+    And I execute on the pod:
+      | bash | -c | conntrack -L \| grep "<%= cb.nodeport %>" |
+    Then the step should succeed
+    And the output should contain:
+      |<%= cb.host_pod1.ip %>|
+
+    #Deleting the udp listener pod which will trigger a new udp listener pod with new IP
+    When I run the :delete client command with:
```
Comment (pruan-rht) [inline features/networking/pod.feature]: @anuragthehatter you should substitute the `:delete` call with `I ensure...`

Comment (anuragthehatter) [reply features/networking/pod.feature]: @pruan-rht Yep. I have used that most of the time. oops how can i forget that.

Comment (anuragthehatter) [reply features/networking/pod.feature]: Fixed and re-tested http://pastebin.test.redhat.com/843972

File: features/networking/pod.feature [hunk 5]
```diff
@@ -298,3 +298,87 @@ Feature: Pod related networking scenarios
...
+
+    # 'yes' command will send a character "h" continously for 3 seconds to /dev/udp on listener where the node is listening for udp traffic on exposed nodeport. The 3 seconds mechanism will create an Assured
+    #  entry which will give us enough time to validate upcoming steps
+    When I run the :exec background client command with:
+      | pod              | <%= cb.client_pod.name %>                             |
+      | oc_opts_end      |                                                       |
+      | exec_command     | bash                                                  |
+      | exec_command_arg | -c                                                    |
+      | exec_command_arg | yes "h">/dev/udp/<%= cb.node_ip %>/<%= cb.nodeport %> |
+    Given 3 seconds have passed
```
Comment (pruan-rht) [inline features/networking/pod.feature]: @anuragthehatter can you pls comment on why the 3 seconds?

Comment (anuragthehatter) [reply features/networking/pod.feature]: @pruan-rht Sure. I already mentioned a comment there. Normally if we send 1 char as a traffic, the conntrack table creates entry for just 30 seconds. So running yes command for 3 seconds is kinda we are making sure we are sending enough data to create ASSURED entry in table for 180 seconds as that will give us enough time to validate further steps. 30 seconds is too risky

Comment (pruan-rht) [reply features/networking/pod.feature]: ok, as long as it works, but kind of hackish...should probably want to look for a more reliable alternative...but ok for now.

Comment (anuragthehatter) [reply features/networking/pod.feature]: yea, this is same as OCP-19810 which have been running fine so far on 3.11 with this logic. So i implemented same for 4.x with extra 4.x requirements. We could have used ncat tool here and on 3.11 case for more reliability but that expects interactive inputs or complicated automation. I can see in future to better implement ncat. Yes is basic unix/linux command comes inbuilt with all kernels so taking just leverage of it.

**Comment by anuragthehatter** [comment]
> @anuragthehatter, you can repelace line 327-331 with `And evalation of `service(cb.host_pod1.name).node_port(port: 8080)` is stored in the :nodeport clipboard`

Thanks @pruan-rht. Great suggestion. Wasn't aware of this def. Shortened the steps and re-ran to make sure http://pastebin.test.redhat.com/845633

**Pull request merged**